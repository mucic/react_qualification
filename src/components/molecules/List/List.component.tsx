import React from 'react'
import * as Styled from "./List.styles";

interface ListProps {
    data: any[]
}

function List(props: ListProps) {

    const {data} = props;
    return (
        <Styled.Container>
                {data.map((repo) => (
                    <li key={repo.id}>
                        <a href={repo.html_url}>{repo.name}</a>
                    </li>
                ))}
        </Styled.Container>
    )
}

export default List
