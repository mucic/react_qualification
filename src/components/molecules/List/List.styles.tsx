import styled from "styled-components";

export const Container = styled.ul`
    width: 100%;
    max-width: 400px;
    list-style: none;
    li {
        margin-bottom: 5px;
        a {
            color: #e30b5d;
            font-family: sans-serif;
            text-decoration: none;
            font-weight: bold;
        }
        &:before {
            content: '\\2713';
            color: #009688;
            font-weight: bold;
            display: inline-block;
            margin-left: -1em;
            width: 1em;
            padding: 10px;
        }
    }
`;