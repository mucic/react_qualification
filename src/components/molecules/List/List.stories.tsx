import React from "react";

import { storiesOf } from "@storybook/react";

import List from "./List.component";
const repos = [
    {
        id: "1",
        html_url: 'www.google.com',
        name: 'Repo No1'
    },
    {
        id: "2",
        html_url: 'www.google.com',
        name: 'Repo No2'
    },
    {
        id: "3",
        html_url: 'www.google.com',
        name: 'Repo No3'
    },
];
storiesOf("List", module).add('defoult', ()=> (
    <List data={repos} />
));