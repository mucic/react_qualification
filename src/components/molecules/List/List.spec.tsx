import React from "react";
import renderer from "react-test-renderer";

import { reposData } from "./List.stories";

import List from "./List.component";

test("List renders as expected", () => {
  const component = renderer.create(
    <List data={reposData} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});