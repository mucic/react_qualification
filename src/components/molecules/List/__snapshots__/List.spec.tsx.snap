// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<ul
  className="sc-AxjAm cMzels"
>
  <li>
    <a
      href="www.google.com"
    >
      Repo No1
    </a>
  </li>
  <li>
    <a
      href="www.google.com"
    >
      Repo No2
    </a>
  </li>
  <li>
    <a
      href="www.google.com"
    >
      Repo No3
    </a>
  </li>
</ul>
`;
